### PR TITLE
feat(ui): introducing dark-mode and refactor some templates and css

### PR DESCRIPTION
**Pull request by ernolf**
Introducing dark-mode

[AIO-dark-mode.webm](https://github.com/user-attachments/assets/f2a6c208-018a-4ac3-9ee0-52dce008df75)

### CODE REVIEW

**Comment by szaimen** [comment]
Woow!!! Many thanks @ernolf for this amazing contribution🎉😊

File: php/templates/already-installed.twig
Comment (szaimen) [inline php/templates/already-installed.twig:19]:
```suggestion
    <div class="login">
        <svg class="nextcloud-logo" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 142 100" width="142" height="100"">
            <use href="/img/nextcloud-logo.svg#logo"></use>
            <use href="/img/nextcloud-logo.svg#Nextcloud"></use>
            <text x="10" y="50" fill="var(--color-nextcloud-logo)" class="fallback-text">Nextcloud Logo</text>
        </svg>
        <h2>Nextcloud All-In-One is already installed</h2>
        <a href="/" class="button" target="_blank" rel="noopener">Open Nextcloud AIO login ↗</a>
    </div>
```

File: php/templates/containers.twig
Comment (szaimen) [inline php/templates/containers.twig:20]: lets please do this in a follow-up PR to keep the changes minimal and related :)

**Review by szaimen — COMMENTED** [review]
I have some remarks though and leave it further down but looks really great already. Thanks again! :)

File: php/public/style.css
Comment (docjyJ) [inline php/public/style.css:419]: Better to use classes instead of identifiers

File: php/public/toggle-dark-mode.js
Comment (docjyJ) [inline php/public/toggle-dark-mode.js:10]: I'm not a fan of emoji, their rendering is random and depends on the system...

File: php/templates/already-installed.twig
Comment (docjyJ) [reply php/templates/already-installed.twig:19]: Yes, and use `a.button` for navigation

File: php/templates/includes/containers-main.twig
Comment (docjyJ) [inline php/templates/includes/containers-main.twig:1]: Can you make 2 commits, 1 for the file naming and 1 for the changes? It will be more readable and will limit conflicts.

File: php/public/toggle-dark-mode.js
Comment (docjyJ) [inline php/public/toggle-dark-mode.js:4]: Would it be possible to add an 'auto' option? Sync with browser theme?

**Review by docjyJ — CHANGES_REQUESTED** [review]
Not tested, but it's cool!

**Comment by docjyJ** [comment]
The ideal would be to have a Vuejs front and a JSON API :face_with_open_eyes_and_hand_over_mouth: :roll_eyes:

**Comment by docjyJ** [comment]
If you don't know, on this site there are lots of tips including sync with OS

https://css-tricks.com/a-complete-guide-to-dark-mode-on-the-web/

**Comment by szaimen** [comment]
> The ideal would be to have a Vuejs front and a JSON API 🫢 🙄

I am not sure about this. I think vuejs is overkill for such a small site. It only adds maintainability issues imho.

File: php/public/toggle-dark-mode.js
Comment (ernolf) [reply php/public/toggle-dark-mode.js:10]: > their rendering is random and depends on the system...

yes and no
Meanwhile the display of emojis is in general as good as uniform in every browser, but some browser extensions (such as Chromoji) can affect the display though. However, since this project already makes a lot of use of emojis and all emoji fonts are already loaded in the CSS and these are two very common emojis that are an integral part of utf8 and are already displayed in every shell, for example, and it doesn't matter at all whether they are displayed exactly in one way or another for this purpose, I wanted to leave it as it is.
I can of course also use two svg elements, but I found that to be absolutely unnecessary considering what I wanted to achieve and display.

File: php/templates/containers.twig
Comment (ernolf) [reply php/templates/containers.twig:20]: OK. :+1:

File: php/templates/includes/containers-main.twig
Comment (ernolf) [reply php/templates/includes/containers-main.twig:1]: Yes, look https://github.com/nextcloud/all-in-one/pull/5397#discussion_r1793551278

File: php/templates/already-installed.twig
Comment (ernolf) [reply php/templates/already-installed.twig]: ✔️done

File: php/templates/includes/containers-main.twig
Comment (ernolf) [reply php/templates/includes/containers-main.twig:1]: ✔️done

File: php/public/toggle-dark-mode.js
Comment (szaimen) [reply php/public/toggle-dark-mode.js:10]: I am fine eith the emoji 👍

File: php/public/toggle-dark-mode.js
Comment (szaimen) [reply php/public/toggle-dark-mode.js:4]: Lets keep the manual mode for the first iteration. We can think about improvements in follow-ups

File: php/templates/containers.twig
Comment (szaimen) [inline php/templates/containers.twig:20]:
```suggestion
            <h1>Nextcloud AIO v9.7.0</h1>
```

File: php/templates/containers.twig
Comment (szaimen) [inline php/templates/containers.twig:44]: Same changes below
```suggestion
                {% if container.GetDisplayName() != '' and class(container.GetRunningState().value == 'running' %}
```

Comment (szaimen) [reply php/templates/containers.twig:44]: you probably need to rebase

**Comment by szaimen** [comment]
Btw @ernolf i think we need to increase for upgrade scenarios: https://github.com/nextcloud/all-in-one/blob/a647d10bf073cc807233290aa3b2cd4b06935699/php/templates/layout.twig#L4

**Comment by docjyJ** [comment]
> I am not sure about this. I think vuejs is overkill for such a small site. And it adds maintainability issues imho.

I just wondered. Why use twig?
Isn't it simpler in standard PHP?

**Comment by szaimen** [comment]
> > I am not sure about this. I think vuejs is overkill for such a small site. And it adds maintainability issues imho.
> 
> I just wondered. Why use twig? Isn't it simpler in standard PHP?

Yes it is :)

**Comment by docjyJ** [comment]
> Yes it is :)

So why twig?

**Comment by szaimen** [comment]
> > Yes it is :)
> 
> So why twig?

Sorry I made a typo. Twig is easier than php :)

**Comment by docjyJ** [comment]
> Sorry I made a typo. Twig is easier than php :)

Maybe... My IDE can't infer types in templates.
What do you use?

**Comment by szaimen** [comment]
> > Sorry I made a typo. Twig is easier than php :)
> 
> Maybe... My IDE can't infer types in templates. What do you use?

Maybe this helps if you should be using phpstorm? https://plugins.jetbrains.com/plugin/7303-twig

**Comment by docjyJ** [comment]
It does not infer type...

**Comment by szaimen** [comment]
> It does not infer type...

Too bad

**Comment by szaimen** [comment]
Lets maybe discuss this in a separate issue. Do you want to create an issue?

**Comment by ernolf** [comment]
> Btw @ernolf i think we need to increase v2 for upgrade scenarios:
> 
> https://github.com/nextcloud/all-in-one/blob/a647d10bf073cc807233290aa3b2cd4b06935699/php/templates/layout.twig#L4

See https://github.com/nextcloud/all-in-one/pull/5397/commits/bf47628073242bd2a045034048e741acbe57b919

**Review by szaimen — APPROVED** [review]
LGTM, thanks a lot 💙
This needs to wait for the next stable release though before it can be merged (ca. 1 week)

File: php/public/style.css
Comment (docjyJ) [reply php/public/style.css:412]: It's OK if is only a sigle element

**Review by docjyJ — APPROVED** [review]
LGTM

File: php/public/toggle-dark-mode.js
Comment (szaimen) [inline php/public/toggle-dark-mode.js:16]: due to the above logic there is no icon shown if you load the page the first time after the update

**Comment by szaimen** [comment]
Another thing: the spinner does not have a dark mode yet and the log out btton should be hidden behind the spinner:
image

**Comment by szaimen** [comment]
Another thing:
it seems like upon reload in dark mode you get a white flashing. not sure how/if we can resolve it but it is quite disturbing:

https://github.com/user-attachments/assets/d12fa7eb-79ed-4388-a215-e9f428fa6ddc

**Comment by szaimen** [comment]
I fear we also need to adjust the mobile layout: the icons on top should not overlay the content and the bottom shuld not leak through:
image

**Comment by szaimen** [comment]
This is now released with v9.8.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Pull request merged**
